eradiate/scenes/atmosphere/homogeneous.py

```diff
@@ -6,84 +6,105 @@
-    validator_is_positive
+    validator_or_auto
 )
-from ...util.units import config_default_units as cdu
-from ...util.units import kernel_default_units as kdu
+from ...util.collections import onedict_value
 from ...util.units import ureg
 
 
-@SceneElementFactory.register("rayleigh_homogeneous")
+@SceneElementFactory.register("homogeneous")
 @attr.s()
-class RayleighHomogeneousAtmosphere(Atmosphere):
-    """Rayleigh homogeneous atmosphere scene element
-    [:factorykey:`rayleigh_homogeneous`].
+class HomogeneousAtmosphere(Atmosphere):
+    """Homogeneous atmosphere scene element [:factorykey:`homogeneous`].
 
-    This class builds an atmosphere consisting of a non-absorbing
-    homogeneous medium. Scattering uses the Rayleigh phase function and the
-    Rayleigh scattering coefficient of air at standard number density (
-    see :func:`sigma_s_air`).
+    This class builds an atmosphere consisting of a homogeneous medium.
+    Scattering uses the Rayleigh phase function.
 
     See :class:`~eradiate.scenes.atmosphere.base.Atmosphere` for undocumented
     members.
 
     .. rubric:: Constructor arguments / instance attributes
 
-    ``sigma_s`` (float or "auto"):
+    ``sigma_s`` (:class:`~eradiate.scenes.spectra.Spectrum` or "auto"):
         Atmosphere scattering coefficient value. If set to ``"auto"``,
         the scattering coefficient will be computed based on the current
         operational mode configuration using the :func:`sigma_s_air`
         function. Default: ``"auto"``.
 
-        Unit-enabled field (default unit: cdu[collision_coefficient]).
+        Can be initialised with a dictionary processed by
+        :class:`.SceneElementFactory`.
+
+    ``sigma_a`` (:class:`~eradiate.scenes.spectra.Spectrum`):
+        Atmosphere absorption coefficient value. Default: 0 (no absorption).
+
+        Can be initialised with a dictionary processed by
+        :class:`.SceneElementFactory`.
 
     """
 
-    sigma_s = attrib_quantity(
+    sigma_s = attr.ib(
         default="auto",
-        converter=converter_or_auto(converter_to_units(cdu.generator("collision_coefficient"))),
-        validator=validator_or_auto(validator_units_compatible(ureg.m ** -1), validator_is_positive),
-        units_compatible=ureg.m ** -1,
-        units_add_converter=False,
-        units_add_validator=False,
-    )  # TODO: turn into a Spectrum
+        converter=converter_or_auto(
+            Spectrum.converter("collision_coefficient")
+        ),
+        validator=validator_or_auto(
+            attr.validators.instance_of(Spectrum),
+            validator_has_quantity("collision_coefficient")
+        ),
+    )
+
+    sigma_a = attr.ib(
+        factory=lambda: UniformCollisionCoefficientSpectrum(value=0.),
+        converter=Spectrum.converter("collision_coefficient"),
+        validator=[attr.validators.instance_of(Spectrum),
+                   validator_has_quantity("collision_coefficient")]
+    )
 
     @property
     def kernel_height(self):
```

Comment:
Don't you think that the API docs look a little funny without the docstrings (same for `kernel_width`) since `kernel_offset` has a docstrings, but not the other `kernel_*` properties?

Reply:
Right. A pity that empty docstrings can't trigger some kind of inheritence from the parent class.